lib/server.js

```diff
@@ -50,7 +51,12 @@ Server.prototype.listen = function listen(port, address, callback) {
 
         var self = this;
 
-        this._socket = dgram.createSocket('udp6');
+        var ip_version = net.isIP(address);
+
+        if (!ip_version)
+            self.emit('error', new Error("Invalid IP address"));
+
+        this._socket = dgram.createSocket((ip_version == 4) ? 'udp4' : 'udp6');
```

Comment:
Can you move this logic to a constant / variable that sits higher up?
No sense to repeat it twice. After that happy to merge. :)